src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -141,7 +145,46 @@ def docker_login_to_registry(login_server):
             "Access to registry was denied. Response code: {}".format(response.status_code))
 
     refresh_token = loads(response.content.decode("utf-8"))["refresh_token"]
+    if only_refresh_token:
+        return refresh_token, ''
 
+    authhost = urlunparse((authurl[0], authurl[1], '/oauth2/token', '', '', ''))
+    if repository is None:
+        scope = 'registry:catalog:*'
+    else:
+        scope = 'repository:' + repository + ':pull'
+    content = {
+        'grant_type': 'refresh_token',
+        'refresh_token': refresh_token,
+        'scope': scope,
+        'service': login_server
+    }
+    response = requests.post(authhost, urlencode(content), headers=headers)
+    access_token = loads(response.content.decode("utf-8"))["access_token"]
+
+    return refresh_token, access_token
+
+def _get_login_refresh_token(login_server):
+    '''Obtains a refresh token from the token server for an AAD-enabled registry.
+    :param str login_server: The registry login server URL to log in to
+    '''
+    refresh_token, _ = _get_login_token(login_server)
+    return refresh_token
+
+def get_login_access_token(login_server, repository=None):
+    '''Obtains an access token from the token server for an AAD-enabled registry.
+    :param str login_server: The registry login server URL to log in to
+    :param str repository: repository for which the access token is requested
+    '''
+    only_refresh_token = False
+    _, access_token = _get_login_token(login_server, only_refresh_token, repository)
+    return access_token
+
+def docker_login_to_registry(login_server):
```

Comment:
>docker_login_to_registry [](start = 4, length = 24)

Can we move all the Docker and refresh token semantics to a _docker_utils.py or something appropriately named file
This is all data plane operations and so would be better to not mix with management plane code.